### PR TITLE
Amend PCPhase operator to support RZ convention

### PR DESCRIPTION
**Pull request by Jaybsoni**
**Context:**
PCPhase gates were designed to be primarily used in the QSVT workflow. Swapping to this convention will allow them to be more directly applied to the workflow without having to keep track of additional global phases.

**Description of the Change:**
In the previous implementation of `PCPhase(phi, dim, wires)`, the first `dim` number of entries along the diagonal in the matrix were phase shifted by `e^(i * phi)` while the remaining `2**n - dim` entries of the diagonal were left alone (set to 1). 

This convention is swapped to one in which the remaining `2**n - dim` entries of the diagonal were set to `e^(-i * phi)`. Note, these two conventions only differ by a global phase. 

**Benefits:**
By absorbing the global phase into the representation, we get a cleaner implementation of QSVT which doesn't have to track the global phase as strictly when converting between representations. This convention is also identical when using trained QSP angles which use RZ rotations as their projector controlled phase gates.

### CODE REVIEW

File: pennylane/ops/qubit/parametric_ops_multi_qubit.py
Comment (soranjh) [inline pennylane/ops/qubit/parametric_ops_multi_qubit.py:1]: Are the existing tests enough, after modification, to fully test this new change?

**Review by obliviateandsurrender — CHANGES_REQUESTED** [review]
Thanks @Jaybsoni! Just leaving a few comments. Another thing I would like to clarify regarding "these two conventions only differ by a global phase" would be that isn't the new `phi` in your implementation twice the `phi` in your old one? Not sure if that causes any problems w.r.t its domain.

File: pennylane/ops/qubit/parametric_ops_multi_qubit.py
Comment (Jaybsoni) [reply pennylane/ops/qubit/parametric_ops_multi_qubit.py:1]: Yes! I am just changing the matrix representation. So just adjusting the tests is enough

File: pennylane/ops/qubit/parametric_ops_multi_qubit.py [hunk 1]
```diff
@@ -558,13 +558,13 @@ class PCPhase(Operation):
 
     basis = "Z"
     grad_method = "A"
-    parameter_frequencies = [(1,)]
+    parameter_frequencies = [(2,)]
```
[the remark this answers is not in the document]
Comment (Jaybsoni) [reply pennylane/ops/qubit/parametric_ops_multi_qubit.py:561]: Previously there was only 1 non identity parameter frequency, now there are two. The gradient tests compare against an analytical gradient I computed by hand. So I am pretty confident parameter shift is correct 👍🏼

File: pennylane/ops/qubit/parametric_ops_multi_qubit.py
[the remark this answers is not in the document]
Comment (Jaybsoni) [reply pennylane/ops/qubit/parametric_ops_multi_qubit.py:1]: Fixed 👍🏼

**Comment by Jaybsoni** [comment]
[sc-34913]

File: tests/ops/qubit/test_parametric_ops.py [hunk 2]
```diff
@@ -2105,13 +2117,17 @@ def test_pcphase_eigvals(self):
...
+        expected = np.array(
+            [np.exp(1j * phi), np.exp(1j * phi), np.exp(-1j * phi), np.exp(-1j * phi)]
+        )
         assert np.allclose(op.eigvals(), expected)
 
         # test arbitrary broadcasted phase shift
         phi = np.array([0.5, 0.4, 0.3])
         op = qml.PCPhase(phi, dim=2, wires=[0, 1])
-        expected = np.array([[np.exp(1j * p), np.exp(1j * p), 1.0, 1.0] for p in phi])
+        expected = np.array(
```
Comment (soranjh) [inline tests/ops/qubit/test_parametric_ops.py:2128]: How this is computed? If computed by hand or it is a trivial solution, might add a comment to clarify.

File: tests/ops/qubit/test_parametric_ops.py [hunk 3]
```diff
@@ -2105,13 +2117,17 @@ def test_pcphase_eigvals(self):
         # test arbitrary phase shift
         phi = 0.5432
         op = qml.PCPhase(phi, dim=2, wires=[0, 1])
-        expected = np.array([np.exp(1j * phi), np.exp(1j * phi), 1.0, 1.0])
+        expected = np.array(
```
Comment (soranjh) [inline tests/ops/qubit/test_parametric_ops.py:2120]: How this is computed? If computed by hand or it is a trivial solution, might add a comment to clarify.

File: tests/ops/qubit/test_parametric_ops.py [hunk 4]
```diff
@@ -1061,7 +1071,9 @@ def test_pcphase_jax(self, phi, dim, wires):
         mat2 = op.compute_matrix(*op.parameters, **op.hyperparameters)
 
         expected_mat = jnp.diag(
-            jnp.array([jnp.exp(1j * phi) if i < dim else 1.0 for i in range(2**num_wires)])
+            jnp.array(
```
Comment (soranjh) [inline tests/ops/qubit/test_parametric_ops.py:1074]: How this is computed? If computed by hand or it is a trivial solution, might add a comment to clarify.

File: tests/ops/qubit/test_parametric_ops.py [hunk 5]
```diff
@@ -1038,7 +1046,9 @@ def test_pcphase_torch(self, phi, dim, wires):
         mat2 = op.compute_matrix(*op.parameters, **op.hyperparameters)
 
         expected_mat = torch.tensor(
-            np.diag([np.exp(1j * phi) if i < dim else 1.0 for i in range(2**num_wires)])
+            np.diag(
```
Comment (soranjh) [inline tests/ops/qubit/test_parametric_ops.py:1049]: How this is computed? If computed by hand or it is a trivial solution, might add a comment to clarify.

File: tests/ops/qubit/test_parametric_ops.py [hunk 6]
```diff
@@ -1017,7 +1023,9 @@ def test_pcphase_tf(self, phi, dim, wires):
         mat2 = op.compute_matrix(*op.parameters, **op.hyperparameters)
 
         expected_mat = tf.Variable(
-            np.diag([np.exp(1j * phi) if i < dim else 1.0 for i in range(2**num_wires)])
+            np.diag(
```
Comment (soranjh) [inline tests/ops/qubit/test_parametric_ops.py:1026]: How this is computed? If computed by hand or it is a trivial solution, might add a comment to clarify.

File: tests/ops/qubit/test_parametric_ops.py [hunk 7]
```diff
@@ -996,7 +1002,7 @@ def test_pcphase(self, phi, dim, wires):
         mat2 = op.compute_matrix(*op.parameters, **op.hyperparameters)
 
         expected_mat = np.diag(
-            [np.exp(1j * phi) if i < dim else 1.0 for i in range(2**num_wires)]
+            [np.exp(1j * phi) if i < dim else np.exp(-1j * phi) for i in range(2**num_wires)]
```
Comment (soranjh) [inline tests/ops/qubit/test_parametric_ops.py:1005]: How this is computed? If computed by hand or it is a trivial solution, might add a comment to clarify.

File: tests/ops/qubit/test_parametric_ops.py [hunk 8]
```diff
@@ -171,7 +171,13 @@ def circuit(phase, dim):
             return qml.state()
 
         def _get_expected_state(phase, dim, size):
-            return np.array([np.exp(1j * phase) if i < dim else 1.0 for i in range(size)]) * 1 / 2
+            return (
```
Comment (soranjh) [inline tests/ops/qubit/test_parametric_ops.py:174]: How this is computed? If computed by hand or it is a trivial solution, might add a comment to clarify.

**Review by soranjh — APPROVED** [review]

**Review by obliviateandsurrender — APPROVED** [review]
Looks good. Is the dipole moment CI failure related to the recent change in pyscf? We had pushed a fix in the master branch for this.

**Comment by Jaybsoni** [comment]
> Looks good. Is the dipole moment CI failure related to the recent change in pyscf? We had pushed a fix in the master branch for this.

I am not sure, let me try merging in master to see if that fixes it?

**Comment by soranjh** [comment]
> > Looks good. Is the dipole moment CI failure related to the recent change in pyscf? We had pushed a fix in the master branch for this.
> 
> I am not sure, let me try merging in master to see if that fixes it?

Yes, this issue has been fixed [here](https://github.com/PennyLaneAI/pennylane/pull/3908).

File: tests/ops/qubit/test_parametric_ops.py (hunk 8 above)
Comment (Jaybsoni) [reply tests/ops/qubit/test_parametric_ops.py:174]: Its trivial, the result is by definition of the operator

File: tests/ops/qubit/test_parametric_ops.py (hunk 7 above)
Comment (Jaybsoni) [reply tests/ops/qubit/test_parametric_ops.py:1005]: Its trivial, the result is by definition of the operator

File: tests/ops/qubit/test_parametric_ops.py (hunk 6 above)
Comment (Jaybsoni) [reply tests/ops/qubit/test_parametric_ops.py:1026]: Its trivial, the result is by definition of the operator

File: tests/ops/qubit/test_parametric_ops.py (hunk 5 above)
Comment (Jaybsoni) [reply tests/ops/qubit/test_parametric_ops.py:1049]: Its trivial, the result is by definition of the operator

File: tests/ops/qubit/test_parametric_ops.py (hunk 3 above)
Comment (Jaybsoni) [reply tests/ops/qubit/test_parametric_ops.py:2120]: Its trivial, the result is by definition of the operator
